### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by illiashcherbak**

### CODE REVIEW

File: src/convertToObject.js [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const convertedStyles = sourceString
+    .split(';')
+    .map(item => item.trim())
+    .filter(item => item !== '')
+    .map(item => item.split(':').map(value => value.trim()))
+    .reduce((prev, curr) => ({
```
Comment (KrisPlanka) [inline src/convertToObject.js:14]: you should move second .map to the new line

**Review by KrisPlanka — APPROVED** [review]
Looks good

File: src/convertToObject.js [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
 function convertToObject(sourceString) {
-  // write your code here
+  const convertedStyles = sourceString
+    .split(';')
+    .map(item => item.trim())
+    .filter(item => item !== '')
+    .map(item => item.split(':')
+      .map(value => value.trim()))
+    .reduce((prev, curr) => ({
+      ...prev,
```
Comment (ivan-kadykalo) [inline src/convertToObject.js:16]: Wow, 5 iterations. Not the best performance, let's optimize this algorithm and made it with only 1 iteration.

Comment (illiashcherbak) [reply src/convertToObject.js]: I was only able to remove one map method. I have no idea how I can do this in one iteration

Comment (ivan-kadykalo) [reply src/convertToObject.js]: @illiashcherbak imagine if you will have such a task, to optimize the productivity of some solution, what will you do? 
Saying "I have no idea" is not the best solution. Why didn't you come to Q&A if you don't know how to do it?

I will wait for you on my Q&A tomorrow at 13:30 and we will discuss how to resolve it in 1 iteration.

**Review by ivan-kadykalo — CHANGES_REQUESTED** [review]
You need to learn how to solve tasks or ask if you don't know how.
Will wait for you on my Q&A tomorrow.

**Pull request closed without merging**